# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ContentParent.cpp
# Commit: 40fcc0cdc593
# Full Hash: 40fcc0cdc593fa337273631de435a2901ec881f6
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-02-07 04:17:40
# Regressor Bug: 1878286
# File Overlap Count: 2
# Description:
#   Bug 1878286 - Eliminate usage of fontlist::Pointer in IPDL. r=gfx-reviewers,lsalzman
#   
#   No change in behavior.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D200564
# ==============================================================================

diff -r c48ca61bc925 -r 40fcc0cdc593 dom/ipc/ContentParent.cpp
--- a/dom/ipc/ContentParent.cpp	Tue Feb 06 10:15:29 2024 +0000
+++ b/dom/ipc/ContentParent.cpp	Tue Feb 06 10:41:50 2024 +0000
@@ -5924,11 +5924,13 @@
 }
 
 mozilla::ipc::IPCResult ContentParent::RecvSetCharacterMap(
-    const uint32_t& aGeneration, const mozilla::fontlist::Pointer& aFacePtr,
+    const uint32_t& aGeneration, const uint32_t& aFamilyIndex,
+    const bool& aAlias, const uint32_t& aFaceIndex,
     const gfxSparseBitSet& aMap) {
   auto* fontList = gfxPlatformFontList::PlatformFontList();
   MOZ_RELEASE_ASSERT(fontList, "gfxPlatformFontList not initialized?");
-  fontList->SetCharacterMap(aGeneration, aFacePtr, aMap);
+  fontList->SetCharacterMap(aGeneration, aFamilyIndex, aAlias, aFaceIndex,
+                            aMap);
   return IPC_OK();
 }
 
@@ -5941,10 +5943,10 @@
 }
 
 mozilla::ipc::IPCResult ContentParent::RecvSetupFamilyCharMap(
-    const uint32_t& aGeneration, const mozilla::fontlist::Pointer& aFamilyPtr) {
+    const uint32_t& aGeneration, const uint32_t& aIndex, const bool& aAlias) {
   auto* fontList = gfxPlatformFontList::PlatformFontList();
   MOZ_RELEASE_ASSERT(fontList, "gfxPlatformFontList not initialized?");
-  fontList->SetupFamilyCharMap(aGeneration, aFamilyPtr);
+  fontList->SetupFamilyCharMap(aGeneration, aIndex, aAlias);
   return IPC_OK();
 }
 